### DISCUSSION TITLE
NACA0012 airfoil at high angle of attack field inversion

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [mdolab/dafoam#477] [General]
Hi,

I am trying to use field inversion for NACA0012 airfoil at 16 degree angle of attack

I got a working mesh, use the SA model (wall resolved completely) and with empty boundary condition for front and back faces, I manage to get convergence.

But when I switch to symmetry boundary condition, I do not get convergence despite trying all settings i.e. changing the discretizatio nschemes, the solution methods and the boundary conditions.

presumably, by setting the symmetry boundary condition, we are telling OpenFOAM that the problem is 3D when in reality, in my case it is only 2D. And at a high angle of attack such as 16 degrees, the flow separation and physics are 3D in nature which becomes hard to solve with just one cell in the z direction. 

Is there no way around the empty boundary condition? What is the reason anyway for not having it?

The only solution that I can think of at the moment is to provide a few cells in the z direction and make it 3D. Perhaps then there would be sufficient domain in the z direction to account for the 3D nature of setting the symmetry boundary condition.

Would you happen to have any other suggestions that I can try?

**Comment by friedenhe** [comment]
Can you share the case? I am curious why the DASimpleFoam did not work with the sym BC. We do not support empty because it will give seg fault. We don't have time to dig into this....

**Reply by uttamcadambi07** [reply]
Sure.

I will share the OpenFOAM case files only because the entire case has a lot of custom implementations. Essentially, my goal was to run an OF only case (similar to runPrimal)  to see if the residuals converge. For each one of my case, I do this to ensure that the baseline design at least converges. When I tried this with the case I am about to share, it failed to converge.

I have tried with empty boundary conditions and achieve convergence of 1E-06. But with symm BC, I need to raise the primalResidualTolerance to as high as 8E-05. While this solves the issue of convergence, my adjoint does not converge (presumably because the adjoint solver requires a well converged primal solution to perform its task)

[AoA_16_NACA0012.zip](https://github.com/mdolab/dafoam/files/12487018/AoA_16_NACA0012.zip)

**Reply by friedenhe** [reply]
I can't run your case using DAFoam. Can you include the runScript.py file?

**Reply by uttamcadambi07** [reply]
Sure. Here it is: [airfoil_naca0012_16deg.zip](https://github.com/mdolab/dafoam/files/12551701/airfoil_naca0012_16deg.zip)
I have implemented my own turbulence model in SpalartAllmarasFieldInversion (it is the same SpalartAllmaras turbulence model from native OF but implemented the same way SpalartAllmarasFv3FieldInversion in DAFoam has been implemented). This is done because Fv3 version resulted in baseline design failing.

**Reply by friedenhe** [reply]
I can reproduce your issue. I suggest: 1. Update your mesh. Please try to use pyHyp to generate high-quality airfoil meshes. Your mesh has a very large non-ortho near the wall, which is bad for y+1 meshes. 2. Please try to use the upwind scheme to converge the flow, and then use it as the initial condition to run the linearUpwind scheme.

**Reply by lnhang0618** [reply]
I alse found the same question in S809 .When the frontAndback set to empty,it is really easy to converge.But when I switch to symmtery boundary,the residual in Uz is really high(more the 1e-3), and the pressure residual will not converge.
[S809_predict_xiaoh.zip](https://github.com/mdolab/dafoam/files/14945139/S809_predict_xiaoh.zip)

**Reply by lnhang0618** [reply]
@friedenhe I'm not sure why changing the boundary condition from empty to symmetry would cause such changes, but adjusting the fvSolution doesn't make it converge. When it converges to a certain extent, there will be significant oscillations.

**Reply by uttamcadambi07** [reply]
I am going to check my case to see what I did. I’ll keep you posted.

**Comment by friedenhe** [comment]
@lnhang0618, have you tried to converge the simulation with a lower angle of attack and then use it as the initial condition to start a high angle of attack simulation? I managed to use this trick to converge other high angle of attack cases.

**Reply by lnhang0618** [reply]
I set a preconditioner for p, increased the nsweep value, and adjusted the relaxation factors of U from 0.7 to 0.3, which led to the convergence of the results. However, these additional settings are not necessary in empty boundary.

**Reply by uttamcadambi07** [reply]
Relaxation factor of 0.3 is very low for velocity. It is not good practice to have it so low, the results will be unreliable in this case. But thanks for the update

**Reply by lnhang0618** [reply]
I believe that the non-physical fluctuations in the Uz direction are the reason for the lack of convergence. I tried reducing the relaxation factor for U to control the magnitude of this part, but perhaps reducing the tolerance for U could achieve a similar effect?

**Reply by uttamcadambi07** [reply]
Ah i see what the issue is now. Yes you have to set Uz tolerance to be high, Typically, although the tolerance in the Z direction is high and you have a high magnitude of Uz, continuity equation will still be enforced since continuity equation requires you to satisfy dw/dz. So long as that is satisfied, you need not worry about the absolute value of the out of plane component of velocity. In DAFoam if i remember correctly, it checks for the tolerance of Ux and p and then proceeds to the next optimization step.

**Reply by lnhang0618** [reply]
It sounds like you recommend setting the tolerance in the z-direction very large? I understand that either an empty-like approach should be used to eliminate the z-direction velocity, or the z-direction velocity should not be updated (such as setting a very large tolerance limit). But I'm not sure if v1812 supports setting tolerance for a specific direction.

**Reply by uttamcadambi07** [reply]
It does! Just set Uz 0.01 or something like that and Ux 1e-5 and Uy 1e-5
